packages/cache/ARCHITECTURE.md

```diff
@@ -0,0 +1,56 @@
+# TODO: title
+
+## Assumptions
+
+- The `CacheRegistry` will be generated by the "Codegen" system where the full
+  set of query keys are known ahead of time. In Athena's case that will mean
+  that inidivual query ids like `query-id-123` will map to **exactly** the return type of that query.
+  Similarly, `cache.get(someEntityId)` will map to **exactly** the correct type of the "merged model". For an example of generating these type registries, see <https://tsplay.dev/NrnDlN>. This gives strong types for entries whose types are statically known. There is no support for strongly typing dynamic entries -- they will be typed as `unknown` and users will have to typecast or equivalent.
+
+## Features
+
+// TODO: rewrite after writing `save` api
+
+- The cache is serializable. This means e.g. it can be written to and restored from persistent storage. It also means that values must be serializable one way or another.
+
+## Implementation Notes
+
+- 🔥🐉 When cache entries (key,value pairs) are added to the cache (i.e. via `tx.commit()`) particular care is needed.
+  1. The main cache is a `Map<String, WeakRef<ValueType>>`.
+  2. There are additional caches that are `Map<string,ValueType>` to handle the
+     Least-Recently-Used (LRU) and Time-to-Live (TTL) expiration features.
```

Comment:
What stops `WeakRef<ValueType>` from being reclaimed?  My understanding of a wekref is it gets garbage collected regardless, but are we saying since it has a reference in another cache record or a component/route has a reference to the value in turn stops weakref object from being garbage collected?

Reply:
Got it, super clear